app/src/main/java/com/tomtom/sdk/examples/navigation/NavigationExamplesActivity.kt

```diff
@@ -11,15 +11,121 @@
+import com.tomtom.sdk.navigation.NavigationOptions
+import com.tomtom.sdk.navigation.ProgressUpdatedListener
+import com.tomtom.sdk.navigation.RoutePlan
+import com.tomtom.sdk.navigation.online.Configuration
+import com.tomtom.sdk.navigation.online.OnlineTomTomNavigationFactory
+import com.tomtom.sdk.navigation.routereplanner.online.OnlineRouteReplannerFactory
+import com.tomtom.sdk.routing.RoutePlanner
+import com.tomtom.sdk.routing.RoutePlanningCallback
+import com.tomtom.sdk.routing.RoutePlanningResponse
+import com.tomtom.sdk.routing.RoutingFailure
+import com.tomtom.sdk.routing.online.OnlineRoutePlanner
+import com.tomtom.sdk.routing.options.Itinerary
+import com.tomtom.sdk.routing.options.RoutePlanningOptions
+import com.tomtom.sdk.routing.options.guidance.GuidanceOptions
+import com.tomtom.sdk.routing.options.guidance.ProgressPoints
+import com.tomtom.sdk.routing.route.Route
+import com.tomtom.sdk.vehicle.Vehicle
 
 class NavigationExamplesActivity : AppCompatActivity() {
+
     private lateinit var binding: ActivityNavigationExamplesBinding
+    private lateinit var routePlanner: RoutePlanner
+    private lateinit var routePlanningOptions: RoutePlanningOptions
+    private lateinit var route: Route
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityNavigationExamplesBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        planRoute()
+    }
+
+    private fun planRoute() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = TOMTOM_API_KEY)
+        val amsterdam = GeoPoint(latitude = 52.37616, longitude = 4.90828)
+        val rotterdam = GeoPoint(latitude = 51.90546, longitude = 4.46662)
+        val itinerary = Itinerary(origin = amsterdam, destination = rotterdam)
+        routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(progressPoints = ProgressPoints.All),
+            vehicle = Vehicle.Car()
+        )
+        routePlanner.planRoute(
+            routePlanningOptions,
+            object : RoutePlanningCallback {
+                override fun onSuccess(result: RoutePlanningResponse) {
+                    route = result.routes.first()
+                    startNavigation()
+                }
+
+                override fun onFailure(failure: RoutingFailure) {
+                    Log.e(TAG, "Unable to calculate a route: " + failure.message)
+                }
+
+                override fun onRoutePlanned(route: Route) = Unit
+            }
+        )
+    }
+
+    private fun createLocationProvider(): LocationProvider {
+        val routeGeoLocations = route.geometry.map { GeoLocation(it) }
+        val simulationStrategy = InterpolationStrategy(routeGeoLocations)
+        return SimulationLocationProvider.create(strategy = simulationStrategy)
+            .also { locationProvider ->
+                locationProvider.enable()
```

Comment:
It should be enabled when the navigation is started

Reply:
It is not enabled by default, we have to manually trigger it.